plugins/openai_plugin.py

```diff
@@ -177,6 +185,88 @@ def request_http(self, query: dict, user_id: int, test_end_time: float = 0):
 
         return result
 
+    def request_batch_http(self, queries, user_id, test_end_time: float = 0):
+        if not isinstance(queries, list):
+            raise TypeError("Queries must be a list. Given : " + str(type(queries)))
+        
+        if len(queries) < 1:
+            raise ValueError("Empty list passed to the request. This is not intended")
+        
+        if len(queries) == 1:
+            UserWarning("Single prompt requests must not be run with the batch API")
+        
+        result = RequestResult(
+            user_id = user_id,
+            input_id = [query.get("input_id") for query in queries],
+            input_tokens = [query.get("input_tokens") for query in queries],
+            )
+
+        headers = {
+            "Content-type": "application/json"
+        }
+
+        if "chat/completions" in self.host:
+            raise ValueError("Batch requests are supported only in v1/completions API")
+        
+        data = {
+            "prompt" : [query.get("text") for query in queries],
```

Comment:
This is not a question of validity but rather the data format coming from the dataset. In the single request format, there is a direct get in the form of "content":query["text"]. This is merely the corresponding version of that.

Reply:
Sorry to clarify I meant this should be fine since assume that the dataset loader validates each query. Just a small detail and not important to fix.

```suggestion
            "prompt" : [query["text"] for query in queries],
```

Reply:
Resolved in [Fix : Potential KeyErrors](https://github.com/openshift-psap/llm-load-test/pull/66/commits/d32611035fdba1e265c270a76a8c1e59da69e7e7)